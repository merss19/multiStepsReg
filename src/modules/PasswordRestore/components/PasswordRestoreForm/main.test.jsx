"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var enzyme_1 = require("enzyme");
var main_1 = require("./main");
var React = require("react");
var enzyme_to_json_1 = require("enzyme-to-json");
var Button_1 = require("../../../../components/Button");
var enzyme_2 = require("enzyme");
var Adapter = require("enzyme-adapter-react-16");
enzyme_2.configure({ adapter: new Adapter() });
var onSubmit;
var handleSubmit;
var start = function () {
    beforeEach(function () {
        onSubmit = jest.fn(),
            handleSubmit = jest.fn();
    });
    var props = {
        onSubmit: onSubmit,
        handleSubmit: handleSubmit
    };
    var buildComponent = function () {
        return enzyme_1.shallow(<main_1.PasswordRestoreForm {...props}/>);
    };
    return {
        buildComponent: buildComponent,
        props: props
    };
};
describe('Component PasswordRestoreForm', function () {
    start();
    it('renders component', function () {
        var component = start().buildComponent();
        expect(component).toHaveLength(1);
    });
    it('renders correctly', function () {
        var component = start().buildComponent();
        var tree = enzyme_to_json_1.default(start().buildComponent());
        expect(tree).toMatchSnapshot();
    });
    it('pass prop onSubmit', function () {
        var component = start().buildComponent();
        expect(component.instance().props.onSubmit).toBe(start().props.onSubmit);
    });
    it('call handleSubmit', function () {
        var component = start().buildComponent();
        component.find(Button_1.Button).simulate('click');
        expect(start().props.handleSubmit).toHaveBeenCalledWith(start().props.onSubmit);
    });
});
