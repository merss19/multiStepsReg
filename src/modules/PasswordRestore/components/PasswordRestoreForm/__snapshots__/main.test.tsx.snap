// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PasswordRestoreForm renders correctly 1`] = `
<div
  className="entry__inner"
>
  <div
    className="entry__header"
  >
    <h2
      className="entry__title text-center"
    >
      Восстановление пароля
    </h2>
  </div>
  <form
    className="entry__box"
  >
    <Field
      component={[Function]}
      id="pass"
      name="pass"
      placeholder="Новый пароль"
    />
    <Field
      component={[Function]}
      id="passAgain"
      name="passAgain"
      placeholder="Новый пароль повторно"
    />
    <Component
      prefix="btn"
      styleBtn="secondary"
      type="submit"
      wide={true}
    >
      Восстановить пароль
    </Component>
    <ul
      className="entry__link text-center"
    >
      <li
        className="entry-nav__item"
      >
        <Link
          replace={false}
          to="/signup"
        >
          Регистрация
        </Link>
      </li>
    </ul>
  </form>
</div>
`;
